js/views/modals/Settings/Page.js

```diff
@@ -37,27 +92,46 @@ export default class extends baseVw {
     this.profile.set(formData);
 
     const save = this.profile.save();
+    let saveAvatar;
+    let saveHeader;
 
     this.trigger('saving');
 
     if (!save) {
       // client side validation failed
       this.trigger('saveComplete', true);
+      this.render();
     } else {
       this.trigger('savingToServer');
-
-      save.done(() => this.trigger('saveComplete'))
-        .fail((...args) =>
-          this.trigger('saveComplete', false, true,
-            args[0] && args[0].responseJSON && args[0].responseJSON.reason || ''));
     }
 
-    // render so errrors are shown / cleared
-    this.render();
+    if (this.avatarChanged && this.avatarCropper.cropit('imageSrc')) {
+      saveAvatar = this.saveAvatar();
+      saveAvatar.done((avatarData) => {
+        // set hash in profile to mirror the server copy
+        this.profile.set('avatarHash', avatarData.hash);
+        app.profile.set('avatarHash', avatarData.hash);
+      });
+    }
 
-    const $firstErr = this.$('.errorList:first');
+    if (this.headerChanged && this.headerCropper.cropit('imageSrc')) {
+      saveHeader = this.saveHeader();
+      saveHeader.done((headerData) => {
+        // set hash in profile to mirror the server copy
+        this.profile.set('headerHash', headerData.hash);
+        app.profile.set('headerHash', headerData.hash);
+      });
+    }
 
-    if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();
+    $.whenAll(save, saveAvatar, saveHeader)
+      .done(() => {
+        this.trigger('saveComplete');
+        this.render();
```

Comment:
Ok, this is how I would do it and it would solve this issue.

This is pseudo code, hasn't been tested, but the gist is that in render you aren't exclusively setting the cropIt image src based off the model. If it had already been set, you are setting it to the existing version, otherwise the model, otherwise nothing.

So, at the top of render (before loadTemplate) put this:

```
    let currAvatar;

    if (this.avatarCropper) {
      currAvatar = this.avatarCropper.cropit('imageSrc');
    }
```

then change the logic where you set the imageSrc to:

```
        if (currAvatar || this.profile.get('avatarHash')) {
          const imageSrc = currAvatar ||
            app.getServerUrl(`ipfs/${this.profile.get('avatarHash')}`);

          this.avatarCropper.cropit('imageSrc', imageSrc);
        } 
```

Now with this code in place, you are free to render() to hide/show errors (how it was before) and you won't lose the existing image state that's already in crop it.

Also, when the uploads complete, there's no need to re-render, because the images are already there. In fact, you can't re-render because you can't just re-render at a random time when the user might be typing in a field.